### ISSUE TITLE
Pets combat bug & Line of sight.

### ISSUE THREAD

**Issue by AcidScytheZ** [SunwellTracker/issues#553]
Decription: 
Well arenas started so are the abusers.

How it works: 

5:30 and forward
https://www.twitch.tv/videos/230178412
Cant be sapped but eating nice one.
Simply people send their pet and before the pet even hit or anything  they  are instantly in combat and cant be sapped so are their teammates.
People are also able to send their pet to person behind pillar which is bug as well.

How it should work: 
They shouldn't enter combat before the pet hit.
Just logged my DK in retail and tested it. You enter in combat when the pet hits its target.
If it doesn't the target its same as to try to autoattack something thats miles away (you get combat visual on your UI but you are not actually in combat.
The problem is that they get in combat when they send the pet from the beginning of the arena.
when we dont talk about pet hit cause its not even the half way to its target and they are both in combat.

In are in retail you also cant send your pet to something that is behind a pillar .
Source (you should point out proofs of your report, please give us some source): 
Can log dk and stream if needed.

Also wotlk retail ESL finals.
https://www.youtube.com/watch?v=rxFIvAv0pWk  08:30-08:45 You can see how warrior charge and pet + the dk are on their way and the dk gets sapped.
https://www.youtube.com/watch?v=kOGyMOAtpIY 1:30 lock cant even send his pet and not only there.

Basically you can check all matches that iNNERFiRE had and see how things work.
That were the top teams during wotlk so far.

If you dont like the quality or something i  make it better its 2009 and if you check other ESL tournaments from that year all have this quality or even 360p

only good quality video is non retail (old ArenaTournament reckful streams )
https://www.youtube.com/watch?v=6qUkfsJ7a0M
As you can see no lock sends his pet to avoid sap.

**Comment by moonsado** [comment]
http://www.arenajunkies.com/topic/118924-combat-bug-with-pet/
http://www.arenajunkies.com/topic/154952-pet-combat-sap-bug/
http://www.arenajunkies.com/topic/72876-pet-combat-bug/
https://github.com/dalaranwow/dalaran-wow/issues/1785

"Basically, while YOU (the pet class) send your pet in, YOU can not be sapped, you're not really in combat, but kinda like a "sub-combat"... "
more stuff in these links.

**Comment by Cwiercpasiec** [comment]
Stream from retail is not really a proof. Also I've just tried to get combat by sending my pet on a mob and while my ghoul was fighting I could  even eat. 
https://imgur.com/a/dEtmv

**Comment by AcidScytheZ** [comment]
Don't fucking link me dalaran WOW! GM wanted LIVE proof so i linked ESL tournament!

**Comment by AcidScytheZ** [comment]
As i can see moonsado has not even checked the videos that i provided.

**Comment by moonsado** [comment]
@Cwiercpasiec yeah thats strange i report that but rip.

**Comment by AcidScytheZ** [comment]
I just cant understand... first they wanted retail proof from back then.
Then i give them proof and they bitch about the quality, excuse me but its rly 2009 and back then majority of people used SQUARE monitors with 1024x768 or 1280x1024 resolutions.

**Comment by AcidScytheZ** [comment]
5:30 and forward
https://www.twitch.tv/videos/230178412
Cant be sapped but eating nice one.
Actually paid 500g to the dk as u can see to come for the test  since the GMs and developers were too busy.

**Comment by ZetoxZ** [comment]
Can't be sapped but eating is correct, that's the same mechanic as why you can mount with your pet in combat but you not.

**Comment by moonsado** [comment]
I made a slightly simliar bug report btw
https://github.com/SunwellTracker/issues/issues/524
Riztazz comment is correct but there is one problem atm I can do op things like cc with my pet and eat/or mount up AS the cc from my pet is still on the target. but ofc when I mount up the pet is gone so is the cc.

**Comment by AcidScytheZ** [comment]
Guys dont talk shit please..... mage water ele uses nova and stuff and i still sap the mage

**Comment by AcidScytheZ** [comment]
So won like 5 games vs Vxi and Pees(dk hpala)(1940 rating) and after that the abuse ofcourse started...
He send his pet and mount up and chase my partner  (pally)
I am done playing versus abusers at 2k mmr.

**Comment by Riztazz** [comment]
I will look into few issues with sending pet on target not in LoS and so on but as far as sapping goes it was like that during wotlk times and it will stay like that for now, unless i get decision from higher-ups that im supposed to fix blizzard mistakes

**Comment by AcidScytheZ** [comment]
@Riztazz did you even spend some time to check 
https://www.youtube.com/watch?v=rxFIvAv0pWk 08:30-08:45

**Comment by AcidScytheZ** [comment]
anyway LoS and Mounting stuff should be nice to see atleast

**Comment by Shatterx** [comment]
Based on comments, we can't be 100% sure. ESL Video vs 2009 Topic
We decided to fix bug with sap, but if someone find irrefutable proof, we'll change our decision.

**Comment by moonsado** [comment]
http://www.arenajunkies.com/topic/103477-3v3-warlock-sap-bug/#entry1375040
http://www.arenajunkies.com/topic/118924-combat-bug-with-pet/
http://www.arenajunkies.com/topic/154952-pet-combat-sap-bug/
http://www.arenajunkies.com/topic/72876-pet-combat-bug/
http://www.arenajunkies.com/topic/143935-saping-locks/

**Comment by Riztazz** [comment]
You're now able to sap a player whos not in combat even if he sent his pet on target, because we decided to resolve that bug that blizzard had in the past. Also resolved an issue where guardians would instantly set victim in combat rather than when they actually hit him, as far as sending pet on target behind LoS, ill write it down as ToDo after our movement rewrite. Closing for now

**Comment by tekarsis** [comment]
I agree with Rgalan89. Even if maybe this mechanic wasn't originally intended to work like this, it was mantained for a long time on retail wow, which means that after all it wasn't considered an issue. Also, as stated above, many other pvp server didn't correct this retail behaviour, and still 2v2 team rogues have had no problem whatsoever reaching the highest arena rankings. So this "fix" in the end may actually give rogues an unfair advantage, since on retail wow their pvp strenghts and weaknesses were shaped along the patches while considering even this behaviour.

 I see the point of creating a better version of wow, and i totally appreciate it, but pvp-wise, in my opinion, is better to keep things like they were, even because any veteran player surely comes from a server where this "fix" never happened and thus could have an hard time adapting to this pvp mechanic, especially if he's playing arena with a pet class. I this there is a space for "missed fixes", and that is quality of life improvements and pve. Pvp, though, is a much more delicate and complex setting, where even a simple mechanic can create imabalances hard to correct later, as huge protests would come from rogues if they get too used with this.

Thank you for yours attention

**Comment by tekarsis** [comment]
I agree with Rgalan89. Even if maybe this mechanic wasn't originally intended to work like this, it was mantained for a long time on retail wow, which means that after all it wasn't considered an issue. Also, as stated above, many other pvp server didn't correct this retail behaviour, and still 2v2 team rogues have had no problem whatsoever reaching the highest arena rankings. So this "fix" in the end may actually give rogues an unfair advantage, since on retail wow their pvp strenghts and weaknesses were shaped along the patches while considering even this behaviour.

The forum posts below, along with many others, decribe that this behaviour was mantained over time. I don't think it is because it never occurred to developers to fix it, but instead because it mantained pvp balancing. For example some classes remove/prevent sap like warrior's enrage, others cancel snares et cetera. It's very likely that preventing sap for pet classes has been seen as a balancing mechanic. Indeed, in a famous pvp server (originated from the previous AT), in the 10 top arena positions there are 3 rogue teams, and just 1 pet class, which shows there haven't been any disadvantage being done to rogues by this mechanic.
https://eu.battle.net/forums/en/wow/topic/17616522235
https://eu.battle.net/forums/en/wow/topic/17615081446
https://eu.battle.net/forums/en/wow/topic/17616973657

I see the point of creating a better version of wow, and i totally appreciate it, but pvp-wise, in my opinion, is better to keep things like they were, also considering that any veteran player surely comes from a server where this "fix" never happened and thus could have an hard time adapting to this mechanic, especially if he's playing arena with a pet class. Instead, from a rogue point of view, this is quite a free and powerful gift. I think there is space for "never implemented fixes", and i think that regards quality of life improvements and pve. Pvp, though, is a much more delicate and complex setting, where even a simple mechanic can create imbalances hard to correct at a later time. For example, in this case, huge protests would come from rogues if they are allowed to get used to this mechanic and suddenly you decide to revert this for one of the reasons explained above.

Thank you for your attention

**Comment by Huntender** [comment]
Please reverse this decision.
